### ISSUE TITLE
IllegalStateException: Not allowed to start service Intent (Android O)

### ISSUE THREAD

**Issue by ligi** [walleth/walleth#106]
```
Android Version: 8.0.0
Phone Model: Nexus 5X
TraceDroid Version: 1.1
TraceDroid extra:
Stacktrace:
java.lang.RuntimeException: Unable to create application org.walleth.App: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.walleth/.core.GethTransactionSigner }: app is in background uid UidRecord{bc7e79 u0a142 RCVR idle procs:1 seq(0,0,0)}
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5794)
at android.app.ActivityThread.-wrap1(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6541)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.walleth/.core.GethTransactionSigner }: app is in background uid UidRecord{bc7e79 u0a142 RCVR idle procs:1 seq(0,0,0)}
at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1505)
at android.app.ContextImpl.startService(ContextImpl.java:1461)
at android.content.ContextWrapper.startService(ContextWrapper.java:644)
at org.walleth.App.executeCodeWeWillIgnoreInTests(App.kt:135)
at org.walleth.App.onCreate(App.kt:87)
at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
... 8 more
```

not yet sure why App.onCreate was called from the background though. As far as I know this is the condition for the crash.

**Comment by ligi** [comment]
Unfortunately the user reporting the problem cannot reproduce it and neither can I currently - tried with a 8.0.0 emulator ..

**Comment by hjubb** [comment]
I had a similar issue with Pixel on 8.1, was able to reproduce it a couple of times by opening the app and immediately putting it into the background. Not sure if this issue has been resolved in an update since then however

**Comment by friedger** [comment]
Should be fixed in 70cd19e4bdce207f5f2dfbeb59fd2fa25f7f2627 (Dec 2017)
